Activities/Cryptography/UiPath.Cryptography/CryptographyHelper.cs

```diff
@@ -45,74 +54,117 @@ public static byte[] HashDataWithKey(KeyedHashAlgorithms keyedHashAlgorithm, byt
+            else
+            {
+#endif
+#if NET5_0
+            if (symmetricAlgorithm == SymmetricAlgorithms.AESGCM)
+            {
+                return EncryptAesGcmNet5(inputBytes, key, out result);
+            }
+            else
+            {
+#endif
+                using (SymmetricAlgorithm algorithm = GetSymmetricAlgorithm(symmetricAlgorithm))
                 {
-                    algorithm.Key = pbkdf2.GetBytes(maxKeySize);
-                }
+                    byte[] encrypted;
+                    byte[] salt = new byte[PBKDF2_SaltSizeBytes];
+                    int maxKeySize = GetLegalKeySizes(algorithm).Max();
 
-                using (ICryptoTransform cryptoTransform = algorithm.CreateEncryptor())
-                {
-                    using (MemoryStream inputStream = new MemoryStream(inputBytes), transformedStream = new MemoryStream())
+                    _rng.GetBytes(salt);
+                    using (Rfc2898DeriveBytes pbkdf2 = new Rfc2898DeriveBytes(key, salt, PBKDF2_Iterations))
                     {
-                        using (CryptoStream cryptoStream = new CryptoStream(inputStream, cryptoTransform, CryptoStreamMode.Read))
+                        algorithm.Key = pbkdf2.GetBytes(maxKeySize);
+                    }
+
+                    using (ICryptoTransform cryptoTransform = algorithm.CreateEncryptor())
+                    {
+                        using (MemoryStream inputStream = new MemoryStream(inputBytes), transformedStream = new MemoryStream())
                         {
-                            cryptoStream.CopyTo(transformedStream);
-                        }
+                            using (CryptoStream cryptoStream = new CryptoStream(inputStream, cryptoTransform, CryptoStreamMode.Read))
+                            {
+                                cryptoStream.CopyTo(transformedStream);
+                            }
 
-                        encrypted = transformedStream.ToArray();
+                            encrypted = transformedStream.ToArray();
+                        }
                     }
+
+                    result = new byte[salt.Length + algorithm.IV.Length + encrypted.Length];
+                    Buffer.BlockCopy(salt, 0, result, 0, salt.Length);
+                    Buffer.BlockCopy(algorithm.IV, 0, result, salt.Length, algorithm.IV.Length);
+                    Buffer.BlockCopy(encrypted, 0, result, salt.Length + algorithm.IV.Length, encrypted.Length);
                 }
 
-                result = new byte[salt.Length + algorithm.IV.Length + encrypted.Length];
-                Buffer.BlockCopy(salt, 0, result, 0, salt.Length);
-                Buffer.BlockCopy(algorithm.IV, 0, result, salt.Length, algorithm.IV.Length);
-                Buffer.BlockCopy(encrypted, 0, result, salt.Length + algorithm.IV.Length, encrypted.Length);
+                return result;
+#if NET461
             }
-
-            return result;
+#endif
+#if NET5_0
+            }
```

Comment:
This kind of preprocessing is really bad practice and is not needed.

You don't need (C#-wise in this case) to have the code like this:

```
if (...) return A
else if (...) return B
else return C
```

You can just have

```
if (...) return A

if (...) return B

return C
```

It makes the code much more readable when preprocessing directives are involved.

Reply:
changed, as requested